### PR TITLE
Add command to reprocess score ranks

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on https://github.com/ppy/osu/pull/30918
- [x] Depends on game package bump w/ above

### CODE REVIEW

**Comment by peppy** [comment]
The code structure is copied from `VerifyImportedScoresCommand` because I thought it would be a good template. Turns out quite a lot changed, but if anything looks weird in the structure of this command then it's likely because it still follows the general structure of that command.

File: osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/VerifyScoreRanksCommand.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+                }
+
+                foreach (var score in scores)
+                {
+                    bool requiresIndexing = false;
+
+                    try
+                    {
+                        var processor = getProcessorForScore(score);
+                        ScoreRank rank = processor.RankFromScore(score.accuracy, score.ScoreData.Statistics);
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/VerifyScoreRanksCommand.cs:114]: > I'm not sure how web can see these proper rounded values either, by the way. Is that elastic...?

I wouldn't be surprised if PHP is just rounding it.

[the remark this answers is not in the document]
Comment (nanaya) [reply osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/VerifyScoreRanksCommand.cs:114]: This one?
```
MySQL [osu]> select * from scores where id = 2261163458\G
*************************** 1. row ***************************
                id: 2261163458
           user_id: 23630277
        ruleset_id: 0
        beatmap_id: 4119149
        has_replay: 0
          preserve: 1
            ranked: 1
              rank: A
            passed: 1
          accuracy: 0.9
         max_combo: 128
       total_score: 463618
              data: {"mods": [], "statistics": {"ok": 64, "meh": 1, "miss": 11, "great": 415, "ignore_hit": 123, "ignore_miss": 7, "small_bonus": 6, "large_tick_hit": 20, "slider_tail_hit": 122}, "maximum_statistics": {"great": 491, "ignore_hit": 124, "large_bonus": 3, "small_bonus": 7, "large_tick_hit": 20, "slider_tail_hit": 124}}
                pp: 91.5358
   legacy_score_id: NULL
legacy_total_score: 0
        started_at: 2024-01-30 13:03:45
          ended_at: 2024-01-30 13:05:12
   unix_updated_at: 1706619912
          build_id: 7400
1 row in set (0.00 sec)
```

[the remark this answers is not in the document]
Comment (bdach) [reply osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/VerifyScoreRanksCommand.cs:114]: I get it now. What we're looking at here is float-to-double promotion.

`RankFromScore()` accepts `double` not `float`: https://github.com/ppy/osu/blob/b71bccc42d8d1e042f3b193e029027b9269a9153/osu.Game/Rulesets/Scoring/ScoreProcessor.cs#L531

And `0.9f` promoted to `double` is exactly what is shown on the OP screenshot: https://sharplab.io/#v2:EYLgHgbALANALiATgVwHYB8ACAmADAWACgcBGIozEgTgAo6ATAe2WABsBTASlwDoqAzTjwAqjAMpxEAS1QBzGp04BuIA

[the remark this answers is not in the document]
Comment (peppy) [reply osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/VerifyScoreRanksCommand.cs:114]: Makes a lot of sense, and also the easiest thing to fix, phew.

**Review by bdach — APPROVED** [review]
seems fine

**Pull request merged**